packages/cli/src/util/dev/server.ts

```diff
@@ -1028,7 +1036,9 @@ export default class DevServer {
 
     await devCommandPromise;
 
-    this.output.ready(`Available at ${link(this.address)}`);
+    // log address without trailing slash to maintian backwards compatibility
+    const addressFormatted = this.addressUrl.toString().replace(/\/$/, '');
+    this.output.ready(`Available at ${link(addressFormatted)}`);
```

Comment:
This part is necessary because if we logged out `this.address.toString()`, there'd be an extra trailing slash. This may not matter, but just in case some is parsing it and expecting consistent output, we should change that output.

Reply:
Is there any chance the address could end up being something like `http://localhost:3000/foo/` where the trailing slash would have meaning?

Reply:
It's technically possible, depending on what value comes in from here: https://github.com/vercel/vercel/blob/a7dbd9649ba21478a1384011a1823573fd93a3b7/packages/cli/src/commands/dev/dev.ts#L30

I added a check that the path is only `/` and tested it out. [`4db6594` (#8538)](https://github.com/vercel/vercel/pull/8538/commits/4db65942e55326d260eeb6d8b7c19f95f8b99577)

Reply:
Hrm. I tried it out and I get an error: `Error: listen ENOTDIR: not a directory localhost:3001/foo/`

But, I like this change anyway in case some refactoring later moves stuff around. This is a more robust replacement of the trailing slassh.

Reply:
There appears to be a lot of different ways to listen, its not always a URL.

It could be `unix:/home/user/server.sock` or `tcp://127.0.0.1:5000

See https://github.com/vercel/vercel/blob/0f720472c4ead9350bc57a190cc5390ee5823d52/packages/cli/test/unit/util/dev/parse-listen.test.ts

Reply:
Ah, neat!

I don't think the dev server will work with non-http anyway, but I can restrict this check to that protocol to be sure. [`8283c6d` (#8538)](https://github.com/vercel/vercel/pull/8538/commits/8283c6db631ac34faedf52f68e5a59434077564a)

Reply:
`unix:` socket _used to_ work, but it seems currently it fails when doing `new URL(this.address)` in various parts of the codebase. I guess we don't have tests for that behavior 🙃